yt/frontends/ytdata/io.py

```diff
@@ -308,12 +316,14 @@ def _read_particle_fields(self, chunks, ptf, selector):
                         yield (ptype, field), data
 
 
-def _get_position_array(ptype, f, ax):
+def _get_position_array(ptype, f, ax, index_mask=None):
+    if index_mask is None:
```

Comment:
I don't actually know either... If you try to modify attributes after instantiating you get errors:

```
a = slice(0, 10)
a.start = 1
```
raises `AttributeError: readonly attribute`

so immutable? maybe? 

I could instead pass in the integer start/end indices here.

Reply:
They aren't, but this follows the idiom we've used in the past.

```python
>>> a = slice(None)
>>> a.start = 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: readonly attribute
```

Reply:
> I would prefer the default value tu be a slice instead of None

I actually started with `def _get_position_array(ptype, f, ax, index_mask=slice(None)):` but one of the linters raised an objection with calling a function within an argument definition.

Reply:
ruff didn't like it
```
ruff.....................................................................Failed
- hook id: ruff
- exit code: 1

yt/frontends/ytdata/io.py:319:50: B008 Do not perform function call `slice` in argument defaults
```

Reply:
Leave it.  I'd prefer we not change the linter *and* violate idioms.

Reply:
Fine by me.